test/unit/os/shell_spec.lua

```diff
@@ -68,5 +68,12 @@ describe('shell functions', function()
       eq(input, output)
       eq(0, status)
     end)
+
+    it ('can return non-zero command execution results', function()
+      local ret = 2
+      local cmd = 'exit' .. ' ' .. ret
```

Comment:
Storing values in variables is good, but it is unnecessary in small scopes like this (where the alternative is just 2 lines). So I removed the variables here.